File: Problem2/Task-5/src/main/resources/Point.py
import math

class Point:
    EPS = 0.0001
    def __init__(self, r, a, polar=False):
        x = r * math.cos(a) if polar else r
        y = r * math.sin(a) if polar else a
        self.x = int(x)
        self.y = int(y)
        if abs(self.x - x) >= Point.EPS or abs(self.y - y) >= Point.EPS:
            raise ValueError("Likely not integers!")
    def getX(self):
        return self.x

    def getY(self):
        return self.y

    def toString(self):
        return f"({self.x}, {self.y})"

File: OriginalTaskSrc/main/java/my/pack/Point.java
package my.pack;

import java.lang.Math;

class Point {
    int x;
    int y;

    Point(int x, int y) {
        this(x, y, false);
    }

    static final double EPS = 0.0001;

    Point(double r, double a/* ngle */, boolean polar) {
        double x = polar ? r * Math.cos(a) : r;
        double y = polar ? r * Math.sin(a) : a;
        this.x = (int) x;
        this.y = (int) y;
        if (Math.abs(this.x - x) >= EPS || Math.abs(this.y - y) >= EPS) {
            throw new IllegalArgumentException("Likely not integers!");
        }
    }

    int getX() {
        return x;
    }

    int getY() {
        return y;
    }

    public String toString() {
        return "(" + x + ", " + y + ")";
    }
}
